protocol/transfer-precompile.md

```diff
@@ -0,0 +1,83 @@
+# Purpose
+
+Since its inception, the Celo blockchain provides a feature called "[Token
+Duality](https://specs.celo.org/token_duality.html)", which allows the native
+token to be used both in the usual ETH-like way and as ERC-20 token.
+
+We would like to bring this feature to more OP-Stack based chains and bring the
+Celo L2 closer to vanilla OP-stack by adding the transfer precompile to
+op-geth.
+
+# Summary
+
+Add a precompile to op-geth that allows transferring native tokens.
+This precompile can only be called by a single privileged contract.
+
+# Problem Statement + Context
+
+Many users and contracts have to deal with both ERC20 tokens and native tokens.
+Since they can't be accessed by the same interface, they either have to be
+handled as two different cases or be explicitly wrapped by an ERC-20 token
+wrapping contract.
+Wrapping tokens is not an ideal solution either, since the wrapped token is
+of limited use (e.g. can't be used to pay gas) and wrapping/unwrapping is
+annoying and incurs additional gas charges.
+
+This can be solved by having a contract that provides an ERC-20 interface
+directly to the native token. To implement such a contract, the blockchain must
+provide a way to transfer native tokens on behalf of the tx sender.
+
+# Proposed Solution
+
+Add one new blockchain parameter `TransferPrecompileAllowed` and a new precompile.
```

Comment:
Do you think calling this `TransferPrecompileAllowedCaller` would make the option more clear?